draft: Migrate to flat configs

jsobacki-revolve opened the pull request:
Migrate to flat configs (support for eslint V9)

Owner has been replaced by the company name (Artur is no longer owner of this repo)
New field "Maintainer" has been added, which I've set to Przemysław Nowak (tech-lead)

Eslint 8 ended support on October 5, 2024
Due to removal of support for .eslintrc.js in eslint 9.x, see:
- https://eslint.org/blog/2022/08/new-config-system-part-1/
- https://eslint.org/blog/2022/08/new-config-system-part-2/
We need to migrate to flat config , so we can continue using eslint >8.x

This PR does change functionality in react-native due to removal of eslint-plugin-react-native (not supported for ~3-4 years) which got replaced by @react-native/eslint-config which is used in our other projects.
Rest of the configuration files should not have any changes in their behaviour.

Since eslint-plugin-node is deprecated, it has been replaced with eslint-plugin-n
eslint-plugin-react-native has been replaced by @react-native/eslint-config since it's no longer maintained by the owner

Version has been bumped to `5.0.0` due to breaking changes

jsobacki-revolve commented:
Przetestuje zarówno React jak i ReactNative w ERP & R&D

można przetestować via: `npm i git+https://git@github.com/revolve-pro/eslint-config-revolve.git#migrate-to-flat-config`

jganczorz-revolve commented:
![image](https://github.com/user-attachments/assets/c553dfc5-f647-4c4e-8ec8-c0d1420a944a)
Co tu się wydarzyło?

jakubsobacki-tl committed:
Migrate to flat configs (support for eslint V9)

Owner has been replaced by the company name (Artur is no longer owner of this repo)
New field "Maintainer" has been added, which I've set to Przemysław Nowak (tech-lead)

Eslint 8 ended support on October 5, 2024
Due to removal of support for .eslintrc.js in eslint 9.x, see:
- https://eslint.org/blog/2022/08/new-config-system-part-1/
- https://eslint.org/blog/2022/08/new-config-system-part-2/
We need to migrate to flat config , so we can continue using eslint >8.x

This PR does change functionality in react-native due to removal of eslint-plugin-react-native (not supported for ~3-4 years) which got replaced by @react-native/eslint-config which is used in our other projects.
Rest of the configuration files should not have any changes in their behaviour.

Since eslint-plugin-node is deprecated, it has been replaced with eslint-plugin-n
eslint-plugin-react-native has been replaced by @react-native/eslint-config since it's no longer maintained by the owner

Version has been bumped to `5.0.0` due to breaking changes

jsobacki-revolve commented:
> Co tu się wydarzyło?

Problemy z ReactNative, zdejmę PR'kę tymczasowo
